source/server/watchdog_impl.h

```diff
@@ -24,15 +24,14 @@ class WatchDogImpl : public WatchDog {
 
   Thread::ThreadId threadId() const override { return thread_id_; }
   // Used by GuardDogImpl determine if the watchdog was touched recently and reset the touch status.
-  bool getTouchedAndReset() { return touched_.exchange(false, std::memory_order_acq_rel); }
+  bool getTouchedAndReset() { return touched_.exchange(false, std::memory_order_relaxed); }
 
   // Server::WatchDog
   void startWatchdog(Event::Dispatcher& dispatcher) override;
   void touch() override {
     // Set touched_ if not already set.
     bool expected = false;
-    touched_.compare_exchange_strong(expected, true, std::memory_order_release,
-                                     std::memory_order_acquire);
+    touched_.compare_exchange_strong(expected, true, std::memory_order_relaxed);
```

Comment:
I don't think you need the CAS here. I think you can just do a relaxed store. It doesn't matter if it's already set?

Reply:
Correct.  I'm using CAS to avoid write in cases where touched_ is already true, since atomic reads are cheaper than atomic writes.

Reply:
OK sounds good.